File: app/models/job.py
from enum import Enum
from pydantic import BaseModel
from typing import Optional

class JobStatus(str, Enum):
    QUEUED = "queued"
    PROCESSING = "processing"
    FINISHED = "finished"
    FAILED = "failed"

class JobCreate(BaseModel):
    filename: str

class JobResponse(BaseModel):
    id: str
    filename: str
    status: JobStatus
    progress: int = 0
    status_text: str = ""
    page_count: Optional[int] = None
    error: Optional[str] = None
    created: str

class HealthResponse(BaseModel):
    ollama_available: bool
    model: str
    model_loaded: bool

File: frontend/src/types/job.ts
export type JobStatus = "queued" | "processing" | "finished" | "failed";

export interface Job {
  id: string;
  filename: string;
  status: JobStatus;
  progress: number;
  status_text: string;
  page_count: number | null;
  error: string | null;
  created: string;
  engine?: string;
}

export interface HealthResponse {
  ollama_available: boolean;
  model: string;
  model_loaded: boolean;
}

export class ApiError extends Error {
  status: number;
  constructor(status: number, message: string) {
    super(message);
    this.status = status;
    this.name = "ApiError";
  }
}
